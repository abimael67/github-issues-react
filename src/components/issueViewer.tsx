import React, { useContext } from "react";
import { Label } from "../types/Issue";
import { IssuesContext } from "../context/issue";

/**Displays issues details */
export const IssueViewer: React.FC = () => {
  const issue = useContext(IssuesContext).selectedIssue;
  return (
    <>
      <div className={"issue-viewer"}>
        <img alt="avatar" src={issue?.user.avatar_url} className="avatar"></img>
        <div className="issue-body">
          <div>{"by " + issue?.user.login}</div>
          <div>{" on " + issue?.created_at}</div>
          <div className="issue-title">{issue?.title}</div>
          <pre>{issue?.body}</pre>
          <div className="labels-display">
            {issue?.labels &&
              issue.labels.map((label: Label, index: number) => (
                <div
                  key={index}
                  title={label.description ?? ""}
                  className="label"
                  style={{ backgroundColor: "#" + label.color }}
                >
                  {label.name}
                </div>
              ))}
          </div>
        </div>
      </div>
    </>
  );
};
