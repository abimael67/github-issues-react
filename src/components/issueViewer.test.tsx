import React from "react";
import renderer from "react-test-renderer";
import { IssueViewer } from "./issueViewer";
import { getFakeIssues } from "../test_utils/fakeApi";

describe("IssueViewer component", () => {
  it("Component renders correctly", () => {
    const tree = renderer
      .create(<IssueViewer issue={getFakeIssues()[0]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
