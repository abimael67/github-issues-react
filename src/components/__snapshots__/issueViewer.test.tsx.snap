// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueViewer component Component renders correctly 1`] = `
<div
  className="issue-viewer"
>
  <img
    alt="avatar"
    className="avatar"
    src="dummy_url"
  />
  <div
    className="issue-body"
  >
    <div>
      by user1
    </div>
    <div>
       on 2020-09-17
    </div>
    <div
      className="issue-title"
    >
      Issue number 1
    </div>
    <pre>
      test body
    </pre>
    <div
      className="labels-display"
    >
      <div
        className="label"
        style={
          Object {
            "backgroundColor": "#ccc",
          }
        }
        title=""
      >
        Label 1
      </div>
    </div>
  </div>
</div>
`;
